horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/data/datastore/ElasticsearchDatastoreConfig.java

```diff
@@ -1,62 +1,44 @@
 package io.hyperfoil.tools.horreum.api.data.datastore;
 
-import static java.util.Objects.requireNonNullElse;
-
 import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import io.hyperfoil.tools.horreum.api.data.datastore.auth.APIKeyAuth;
+import io.hyperfoil.tools.horreum.api.data.datastore.auth.NoAuth;
+import io.hyperfoil.tools.horreum.api.data.datastore.auth.UsernamePassAuth;
 
 @Schema(type = SchemaType.OBJECT, required = true, description = "Type of backend datastore")
 public class ElasticsearchDatastoreConfig extends BaseDatastoreConfig {
 
+    public static final String[] auths = { NoAuth._TYPE, APIKeyAuth._TYPE, UsernamePassAuth._TYPE };
+    public static final String name = "Elasticsearch";
+    public static final String label = "Elasticsearch";
+    public static final Boolean builtIn = false;
+
     public ElasticsearchDatastoreConfig() {
-        super(false);
-    }
 
-    @Schema(type = SchemaType.STRING, description = "Elasticsearch API KEY")
-    public String apiKey;
+    }
 
     @Schema(type = SchemaType.STRING, required = true, description = "Elasticsearch url")
     public String url;
 
-    @Schema(type = SchemaType.STRING, description = "Elasticsearch username")
-    public String username;
-
-    @Schema(type = SchemaType.STRING, description = "Elasticsearch password")
-    @JsonIgnore
-    public String password;
-
-    @JsonProperty("password")
-    public void setSecrets(String password) {
-        this.password = password;
-    }
-
-    @JsonProperty("password")
-    public String getMaskedSecrets() {
-        if (this.password != null) {
-            return "********";
-        } else {
-            return null;
-        }
-    }
-
     @Override
     public String validateConfig() {
 
-        String _apiKey = requireNonNullElse(apiKey, "");
-        String _username = requireNonNullElse(username, "");
-        String _password = requireNonNullElse(password, "");
-
-        if ("".equals(_apiKey) && ("".equals(_username) || "".equals(_password))) {
-            return "Either apiKey or username and password must be set";
+        //TODO:: replace with pattern matching after upgrading to java 17
+        if (authentication instanceof APIKeyAuth) {
+            APIKeyAuth apiKeyAuth = (APIKeyAuth) authentication;
+            if ("".equals(apiKeyAuth.apiKey)) {
+                return "apiKey must be set";
+            }
+        } else if (authentication instanceof UsernamePassAuth) {
+            UsernamePassAuth usernamePassAuth = (UsernamePassAuth) authentication;
+
+            if (usernamePassAuth.username == null || usernamePassAuth.password == null
+                    || "".equals(usernamePassAuth.username) || "".equals(usernamePassAuth.password)) {
```

Comment:
```suggestion
            if (usernamePassAuth.username == null || usernamePassAuth.username.isEmpty()
                    || usernamePassAuth.password == null || usernamePassAuth.password.isEmpty()) {
```

Reply:
will change to `isBlank()` to take into account white-space as well